### PR TITLE
Vendor and build gcs-fetcher image in release

### PR DESCRIPTION
**Pull request by ImJasonH**
This changes the default value of the gcs-build resource image to an
image vendored into the pipelines repo and built during release (or ko
apply for local development).

This gives us more control over the version used by users, since the
image will be built by us from vendored/versioned source.

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ n/a ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ n/a ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ y ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) and [TaskRun](../tekton/publish-run.yaml) to build and release this image

I _think_ this doesn't apply to this PR, but someone who knows more should probably double-check.

# Release Notes

```
The image used to fetch source for `build-gcs` resources is vendored and versioned along with Tekton Pipelines components.
```

### CODE REVIEW

**Comment by bobcatfish** [comment]
(also testing my prow update.... 😅 )

**Comment by bobcatfish** [comment]
Wait can I only re-run failed stuff? I don't even know how this works....

**Comment by vdemeester** [comment]
@ImJasonH 

```
I0805 20:33:53.763]     - lastTransitionTime: "2019-08-05T20:23:08Z"
I0805 20:33:53.763]       message: 'build step "step-storage-fetch-rules-n8kk7" is pending with reason
I0805 20:33:53.763]         "Failed to apply default image tag \"github.com/GoogleCloudPlatform/cloud-builders/gcs-fetcher/cmd/gcs-fetcher\":
I0805 20:33:53.764]         couldn''t parse image reference \"github.com/GoogleCloudPlatform/c
```

File: config/controller.yaml [hunk 1]
```diff
@@ -41,6 +41,8 @@ spec:
           "-entrypoint-image", "github.com/tektoncd/pipeline/cmd/entrypoint",
           "-imagedigest-exporter-image", "github.com/tektoncd/pipeline/cmd/imagedigestexporter",
           "-pr-image", "github.com/tektoncd/pipeline/cmd/pullrequest-init",
+          "-build-gcs-fetcher-image", "github.com/tektoncd/pipeline/vendor/github.com/GoogleCloudPlatform/cloud-builders/gcs-fetcher/cmd/gcs-fetcher",
+        "-build-gcs-uploader-image", "github.com/tektoncd/pipeline/vendor/github.com/GoogleCloudPlatform/cloud-builders/gcs-fetcher/cmd/gcs-uploader",
```
Comment (bobcatfish) [inline config/controller.yaml]: I think there's an indentation inconsistency?

**Comment by ImJasonH** [comment]
/hold cancel

Spacing issue is resolved.

**Pull request merged**